File: daily-coding-problems/2/main.ts
function prodOfAllBrute(arr: number[]): number[]
{
    const a = [];

    for(let i = 0; i < arr.length; i++) {
        let x = 1;

        for(let j = 0; j < arr.length; j++) {
            if(arr[i] == arr[j]) {
                continue;
            }

            x *= arr[j];
        }

        a.push(x);
    }

    return a;
}


function prodOfAll(arr: number[]): number[]
{
    const q = arr.length,
        a = new Array(q),
        b = new Array(q);

    let p = 1;
    for(let i = 0; i < q; i++)
    {
        a[i] = p;
        p *= arr[i];
    }

    p = 1;
    for(let j = (q - 1); j >= 0; j--)
    {
        b[j] = p;
        p *= arr[j];
    }

    const r = [];
    for(let k = 0; k < q; k++)
        r.push(a[k] * b[k]);

    return r;
}


const L = [1, 2, 3, 4, 5];

console.log(prodOfAll(L));

File: daily-coding-problems/2/main.py
#!/usr/bin/env python3


def prod_of_all_brute(arr):
    a = []

    for i in arr:
        x = 1
        for j in arr:
            if i is j:
                continue

            x *= j

        a.append(x)

    return a


def prod_of_all(arr):
    q = len(arr)
    a = [None] * q
    b = [None] * q

    p = 1
    for i in range(0, q):
        a[i] = p
        p *= arr[i]

    p = 1
    for j in range((q - 1), -1, -1):
        b[j] = p
        p *= arr[j]

    r = []
    for k in range(0, q):
        r.append(a[k] * b[k])

    return r


L = [1, 2, 3, 4, 5]


print(prod_of_all(L))
